sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -56,16 +69,33 @@ string json = @"[{""LatinName"":""Animalia"",""Weight"":1.1,""Name"":""Doggo"","
 DogListProperty dog = ModelSerializer.Deserialize<DogListProperty>(BinaryData.FromString(json), options);
 ```
 
-## Using ModelJsonConverter for JsonSerializer
+## Using explicit cast
+
+If you would like to convert the protocol model to the strongly typed model, you can use the explicit cast operator. This will allow you to use the strongly typed model for serialization and deserialization. There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
 
-In order to better integrate with the rest of the .NET ecosystem, Azure.Core supports System.Text.Json serialization. The following example demonstrates using System.Text.Json for serialization and deserialization
-If we go this route the IJsonSerializable interface will only be needed for compile time constraints and can most likely be methodless and renamed to IRehydratable.
+### Serialization
 
-One limitation if we go this route is there isn't a clear place to pass in a flag to include additional properties during serialization and deserialization.
+```C# Snippet:ExplicitCast_Serialize
+DefaultAzureCredential credential = new DefaultAzureCredential();
+PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), credential);
+DogListProperty dog = new DogListProperty("myPet");
+Response response = client.CreatePet("myPet", (RequestContent)dog);
```

Comment:
KrzysztofCwalina 

I would add a comment explaining why one line used a cast and the other uses Create